pkg/server/http.go

```diff
@@ -184,16 +190,43 @@ func httpResponseModifier(ctx context.Context, w http.ResponseWriter, _ proto.Me
 	}
 
 	// set http status code
-	if vals := md.HeaderMD.Get(httpStatusHeader); len(vals) > 0 {
+	if vals := md.HeaderMD.Get(httpStatusCodeHeader); len(vals) > 0 {
 		code, err := strconv.Atoi(vals[0])
 		if err != nil {
 			return err
 		}
 		// delete the headers to not expose any grpc-metadata in http response
-		delete(md.HeaderMD, httpStatusHeader)
+		delete(md.HeaderMD, httpStatusCodeHeader)
 		delete(w.Header(), "Grpc-Metadata-X-Http-Code")
 		w.WriteHeader(code)
 	}
 
+	// set the response body if needed
+	if vals := md.HeaderMD.Get(httpErrorMessageHeader); len(vals) > 0 {
+		msg := vals[0]
+		delete(md.HeaderMD, httpErrorMessageHeader)
+		delete(w.Header(), "Grpc-Metadata-X-Http-Error-Message")
+		_, err := w.Write([]byte(msg + "\n"))
+		if err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
+
+func methodNotAllowedHandler(ctx context.Context, mux *runtime.ServeMux, marshaler runtime.Marshaler, w http.ResponseWriter, r *http.Request, err error) {
```

Comment:
Can you add a comment for why we need this, that we're remapping a gRPC code to a different HTTP code?